core/builder/src/main/java/io/quarkus/builder/BuildChain.java

```diff
@@ -102,7 +102,7 @@ int getInitialMultiCount() {
         return initialMultiCount;
     }
 
-    List<StepInfo> getStartSteps() {
+    Set<StepInfo> getStartSteps() {
```

Comment:
No, it is not required to solve the reproducibility problem, it just simplifies the flow that existed even before this change: the `startSteps` field was initialized from a Set anyway. Now the Set is kept instead of being copied to a new ArrayList. Do you prefer to get back to a List for some reason?

Reply:
No, I suppose it doesn't matter if the steps were already a set.